web/src/main/java/com/graphhopper/http/GraphHopperManaged.java

```diff
@@ -41,8 +52,44 @@ public GraphHopperManaged(CmdArgs configuration) {
         graphHopper = new GraphHopperOSM(
                 SpatialRuleLookupHelper.createLandmarkSplittingFeatureCollection(configuration.get(Parameters.Landmark.PREPARE + "split_area_location", ""))
         ).forServer();
+        graphHopper.setFlagEncoderFactory(new FlagEncoderFactory() {
+            private FlagEncoderFactory delegate = new DefaultFlagEncoderFactory();
+            @Override
+            public FlagEncoder createFlagEncoder(String name, PMap configuration) {
+                if (name.equals("original-direction")) {
+                    return new OriginalDirectionFlagEncoder();
+                }
+                return delegate.createFlagEncoder(name, configuration);
+            }
+        });
         SpatialRuleLookupHelper.buildAndInjectSpatialRuleIntoGH(graphHopper, configuration);
         graphHopper.init(configuration);
+        graphHopper.setPathDetailsBuilderFactory(new PathDetailsBuilderFactory() {
+            @Override
+            public List<PathDetailsBuilder> createPathDetailsBuilders(List<String> requestedPathDetails, FlagEncoder encoder, Weighting weighting) {
+                // request-scoped
+                OriginalDirectionFlagEncoder originalDirectionFlagEncoder = (OriginalDirectionFlagEncoder) graphHopper.getGraphHopperStorage().getEncodingManager().getEncoder("original-direction");
+                List<PathDetailsBuilder> pathDetailsBuilders = super.createPathDetailsBuilders(requestedPathDetails, encoder, weighting);
+                pathDetailsBuilders.add(new AbstractPathDetailsBuilder("r5_edge_id") {
+                    private int edgeId = -1;
+
+                    @Override
+                    public boolean isEdgeDifferentToLastEdge(EdgeIteratorState edge) {
+                        if (edge.getEdge() != edgeId) {
```

Comment:
This won't work. You need to calculate your magic edgeId before comparing the two ids. This case will probably almost never pop up though.

Reply:
So you need something like:
```
int newEdgeId = edgeId = edge.getEdge() * 2 + (originalDirectionFlagEncoder.isOriginalDirection(edge.getFlags()) ? 0 : 1);
if(newEdgeId!=edgeId)
```

Reply:
Or use a different value for the currentValue or even always execute this code (ie. always return true) as we can assume that we loop through a list of edges and always call a new edge ... can or should we assume this?